skglm/datafits/single_task.py

```diff
@@ -712,3 +732,21 @@ def _AT_dot_vec(self, vec):
             out[current_H_idx] = weighted_sum_vec_H * np.ones(size_current_H)
 
         return out
+
+    def _get_indptr(self, vals, indices):
+        # given `indices = argsort(vals)`
+        # build and array `indptr` where two consecutive elements
+        # delimit indices with the same val
+        n_indices = indices.shape[0]
+
+        indptr = [0]
+        count = 1
+        for i in range(1, n_indices):
```

Comment:
is it not more natural to have range(n_indices - 1) and then use i and i + 1 L745?